src/chocolatey/infrastructure.app/nuget/NugetCommon.cs

```diff
@@ -451,16 +451,27 @@ ISet<SourcePackageDependencyInfo> dependencyInfos
 
             //if (availablePackages.Contains(packageID)) return;
 
-            var dependencyInfoResources = resources.DependencyInfoResources();
-
-            foreach (var dependencyInfoResource in dependencyInfoResources)
+            foreach (var resource in resources)
             {
-                IEnumerable<SourcePackageDependencyInfo> dependencyInfos = Array.Empty<SourcePackageDependencyInfo>();
+                var dependencyInfoResource = resource.DependencyInfoResource;
+
+                if (dependencyInfoResource is null)
+                {
+                    // We can't lookup any dependencies using this resource.
+                    continue;
+                }
+
+                // check if we already have packages matching the constraints in the list
+                IEnumerable<SourcePackageDependencyInfo> dependencyInfos = availablePackages.Where(

```

Comment:
We can use `var` here since we know we will be returned a `List<SourcePackageDependencyInfo>`
```suggestion
                var dependencyInfos = availablePackages.Where(
```

Reply:
ResolvePackages returns IEnumerable<SourcePackageDependencyInfo>.  To keep consistency in types, this needs to be IEnumerable<SourcePackageDependencyInfo> or there needs to be a ToList on the results from ResolvePackages (line 472).  Is ToList preferable?

Reply:
Looking closer at the changeset, and slightly beyond the changeset, I see that in the original code it was also being created as `IEnumerable`, so I think it's fine to leave it here. I do question the `ToList()` on the next line, but will open a new comment to discuss that.

Reply:
I've unresolved this conversation and the .Any one below. After some discussions internally, we may still want to make this change, with a few extras. There is nothing that needs actioning on this right now, but we may consider changes prior to merging.